mcp/mcp-server/src/main/java/software/amazon/smithy/java/mcp/server/McpServerProxy.java

```diff
@@ -23,6 +23,7 @@ public abstract class McpServerProxy {
     private static final AtomicInteger ID_GENERATOR = new AtomicInteger(0);
 
     protected Consumer<JsonRpcResponse> notificationConsumer;
+    protected String protocolVersion;
```

Comment:
This needs to be volatile, other wise its not guaranteed to be visible to other threads in subsequent requests.